pkg/stanza/operator/input/windows/subscription.go

```diff
@@ -76,14 +88,47 @@ func (s *Subscription) Read(maxReads int) ([]Event, error) {
 		return nil, fmt.Errorf("max reads must be greater than 0")
 	}
 
+	events, err := s.readWithRetry(maxReads)
+	if s.batchSizeReduced {
+		s.batchSizeReduced = false
+		return events, ErrBatchSizeReduced
+	}
+
+	if err != nil {
+		return nil, err
+	}
+
+	return events, nil
+}
+
+// readWithRetry will read events from the subscription with dynamic batch sizing if the RPC_S_INVALID_BOUND error occurs.
+func (s *Subscription) readWithRetry(maxReads int) ([]Event, error) {
 	eventHandles := make([]uintptr, maxReads)
 	var eventsRead uint32
+
 	err := evtNext(s.handle, uint32(maxReads), &eventHandles[0], 0, 0, &eventsRead)
 
 	if errors.Is(err, ErrorInvalidOperation) && eventsRead == 0 {
 		return nil, nil
 	}
 
+	if err != nil && errors.Is(err, windows.RPC_S_INVALID_BOUND) {
```

Comment:
nit: consistency with previous code 
```suggestion
	if errors.Is(err, windows.RPC_S_INVALID_BOUND) {
```